### PR TITLE
Add coinasl

### PR DESCRIPTION
**Pull request by jschueller**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by saraedum** [comment]
@jschueller are there any issues with the license situation here? It's confusing that the build downloads a tarball from github that is not part of the release itself. What's behind the note `Note: It is YOUR RESPONSIBILITY to ensure that you are entitled to download and use this third party package.` in the project's README?

**Comment by jschueller** [comment]
you're right this is unclear.
the coinasl itself is EPL-1.0
the asl library comes from https://github.com/ampl/mp which is MIT
in turn, asl is meant to be used with proprietary solvers, for which this note is about I think

I'm not interested in these solvers, I just need coinasl to be there because it seems to be a build requirement for coinbonmin

**Comment by saraedum** [comment]
Thanks for clarifying.

Could we point out that this has a mixed license in the recipe?

Also, I am not sure about the download in the recipe since this part of the code ignores the `sha256` in the recipe. Have you looked into https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-multiple-sources for this?

**Comment by jschueller** [comment]
I added the file and mentionned that asl is MIT
It's less practical to download solvers from meta.yaml because a patch is applied.

**Comment by saraedum** [comment]
I don't understand your comment about the patches. Are you referring to the two patches that are applied by the download script? Couldn't you apply them as part of the `build.sh`?

Please understand that I am not part of the conda-forge core team, so I do not feel sufficiently senior to wave this through since it downloads code for the build.

**Comment by jschueller** [comment]
I guess it could be possible, but I prefer to keep calling their script if they add some patches etc

File: recipes/coinasl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib/libcoinasl${SHLIB_EXT}"
+
+about:
+  home: http://www.coin-or.org/
+  license: EPL-1.0
+  # additional MIT license for ASL code
+  license_file: LICENSE.rst
```
Comment (saraedum) [inline recipes/coinasl/meta.yaml:39]: I think you can package more than one license file here: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file

Comment (jschueller) [reply recipes/coinasl/meta.yaml:39]: indeed

**Comment by jschueller** [comment]
turns ou we can use ampl-mp package directly

**Pull request closed without merging**